api/controllers/documentSection.js

```diff
@@ -73,45 +81,85 @@ exports.protectedGet = async function (args, res) {
   }
 };
 
-
 /**
  * Save a new document section.
  * 
  * @param {object} args The arguments used to save the section.
  * @param {HTTPResponse} res The response used for the HTTP route.
  * @returns {object}
  */
-exports.protectedPost = async function (args, res) {
-    defaultLog.info('DOCUMENT SECTION PROTECTED POST');
-    const obj = args.swagger.params.documentSection.value;
-  
-    defaultLog.info('Incoming new document section:', obj);
-  
-    const DocumentSection = mongoose.model('DocumentSection');
-  
-    const documentSection = new DocumentSection({
-      _schemaName: 'DocumentSection',
-      name: obj.name,
-      project: obj.project,
-      order: obj.order,
-  
-      read: ['staff', 'sysadmin'],
-      write: ['staff', 'sysadmin'],
-      delete: ['staff', 'sysadmin']
-    });
-  
-    try {
-      const docSectionResult = await documentSection.save();
-      Utils.recordAction(
-        'Post',
-        'DocumentSection',
-        args.swagger.params.auth_payload.preferred_username,
-        docSectionResult._id
-        );
-      defaultLog.info('Saved new document section object:', docSectionResult._id);
-      return Actions.sendResponse(res, 200, docSectionResult);
-    } catch (e) {
-      defaultLog.error(e);
-      return Actions.sendResponse(res, 400, e);
+exports.protectedPost = async (args, res) => {
+  defaultLog.info('DOCUMENT SECTION PROTECTED POST');
+  const obj = args.swagger.params.documentSection.value;
+
+  defaultLog.info('Incoming new document section:', obj);
+
+  const DocumentSection = mongoose.model('DocumentSection');
+
+  const documentSection = new DocumentSection({
+    _schemaName: 'DocumentSection',
+    name: obj.name,
+    project: obj.project,
+    order: obj.order,
+
+    read: ['staff', 'sysadmin'],
+    write: ['staff', 'sysadmin'],
+    delete: ['staff', 'sysadmin']
+  });
+
+  try {
+    const docSectionResult = await documentSection.save();
+    Utils.recordAction(
+      'Post',
+      'DocumentSection',
+      args.swagger.params.auth_payload.preferred_username,
+      docSectionResult._id
+      );
+    defaultLog.info('Saved new document section object:', docSectionResult._id);
```

Comment:
i'm assuming you know this will be ok, because every possible combination except a save will throw an error.

Reply:
Thanks for the feedback. When you say "combination" are you referring to other DB operations in this controller file? 

I believe this particular operation should fail gracefully due to being nested in the try/catch. This pattern is idiomatic to the API codebase